File: problems/0001-two-sum/cpp/test_solution.cpp
#include <cassert>
#include <iostream>
#include <vector>

#include "solution.cpp"

using namespace std;

int main() {
  Solution solver;

  vector<int> nums1 = {2, 7, 11, 15};
  vector<int> res1 = solver.twoSum(nums1, 9);
  assert((res1 == vector<int>{0, 1}));

  vector<int> nums2 = {3, 2, 4};
  vector<int> res2 = solver.twoSum(nums2, 6);
  assert((res2 == vector<int>{1, 2}));

  vector<int> nums3 = {3, 3};
  vector<int> res3 = solver.twoSum(nums3, 6);
  assert((res3 == vector<int>{0, 1}));

  cout << "All C++ tests passed for Two Sum!\n";
  return 0;
}

File: problems/0001-two-sum/python/test_solution.py
"""Test cases for Two Sum problem."""

from solution import Solution


def test_two_sum():
    """Test the twoSum method with various inputs."""
    solver = Solution()
    assert solver.twoSum([2, 7, 11, 15], 9) == [0, 1]
    assert solver.twoSum([3, 2, 4], 6) == [1, 2]
    assert solver.twoSum([3, 3], 6) == [0, 1]
    print("All Python tests passed for Two Sum.")


if __name__ == "__main__":
    test_two_sum()
